File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer,Integer> st = new HashMap<>();
        int[] ans = new int[2];
        
        for(int i = 0; i<nums.length; i++){
            int x = target-nums[i];
            if(st.containsKey(x)){
                ans[0] = i;
                ans[1] = st.get(x);
            }
            st.put(nums[i],i);
        }
        return ans;
    }
}

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        st = {}

        for i in range(len(nums)):
            if(target - nums[i]) in st:
                return [st[target - nums[i]],i]
            else:
                st[nums[i]] = i

        return [] 